LanguageFeatures/Parts-with-imports/scope_A08_t02.dart

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It’s a compile-time error if any file declares an import prefix
+/// with the same base name as a top-level declaration of the library.
+///
+/// @description Check that it is a compile-time error if a part-file declares
+/// an import prefix with the same base name as a top-level declaration.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=enhanced-parts
+
+part 'scope_A08_t01_part1.dart';
+part 'scope_A08_t01_branch2_part1.dart';
```

Comment:
I think this should be moved after imports, and adjusted:
```suggestion
part 'scope_A08_t02_part1.dart';
part 'scope_A08_t02_branch2_part1.dart';
```

Reply:
This seemed useful:

```mermaid
graph BT;
  lib82["scope_A08_t02.dart"] -->|as l*| scopeLib2["scope_lib2.dart"]
  part821 -->|as p*| scopeLib2
  part821["scope_A08_t02_part1.dart"] -->|part of| lib82
  part82b1["scope_A08_t02_branch2_part1.dart"] -->|part of| lib82
  part82b2["scope_A08_t02_branch2_part2.dart"] -->|part of| part82b1
  part822["scope_A08_t02_part2.dart"] -->|part of| part821
```
```
graph BT;
  lib82["scope_A08_t02.dart"] -->|as l*| scopeLib2["scope_lib2.dart"]
  part821 -->|as p*| scopeLib2
  part821["scope_A08_t02_part1.dart"] -->|part of| lib82
  part82b1["scope_A08_t02_branch2_part1.dart"] -->|part of| lib82
  part82b2["scope_A08_t02_branch2_part2.dart"] -->|part of| part82b1
  part822["scope_A08_t02_part2.dart"] -->|part of| part821
```